system/jlib/jlog.hpp

```diff
@@ -796,23 +806,23 @@ extern jlib_decl ILogMsgHandler * attachLogMsgMonitorFromPTree(IPropertyTree * t
 extern jlib_decl void attachManyLogMsgMonitorsFromPTree(IPropertyTree * tree);            // Takes tree containing many <monitor> elements
 
 // Standard categories and unknown jobInfo
-constexpr LogMsgCategory MCdisaster(MSGAUD_all, MSGCLS_disaster);
-constexpr LogMsgCategory MCuserError(MSGAUD_user, MSGCLS_error);
-constexpr LogMsgCategory MCoperatorError(MSGAUD_operator, MSGCLS_error);
-constexpr LogMsgCategory MCinternalError(MSGAUD_programmer, MSGCLS_error, 1);
-constexpr LogMsgCategory MCauditError(MSGAUD_audit, MSGCLS_error);
-constexpr LogMsgCategory MCuserWarning(MSGAUD_user, MSGCLS_warning);
-constexpr LogMsgCategory MCoperatorWarning(MSGAUD_operator, MSGCLS_warning);
-constexpr LogMsgCategory MCinternalWarning(MSGAUD_programmer, MSGCLS_warning, 1);
-constexpr LogMsgCategory MCauditWarning(MSGAUD_audit, MSGCLS_warning);
-constexpr LogMsgCategory MCuserProgress(MSGAUD_user, MSGCLS_progress);
-constexpr LogMsgCategory MCoperatorProgress(MSGAUD_operator, MSGCLS_progress);
-constexpr LogMsgCategory MCdebugProgress(MSGAUD_programmer, MSGCLS_progress);
-constexpr LogMsgCategory MCuserInfo(MSGAUD_user, MSGCLS_information);
-constexpr LogMsgCategory MCdebugInfo(MSGAUD_programmer, MSGCLS_information);
-constexpr LogMsgCategory MCauditInfo(MSGAUD_audit, MSGCLS_information);
-constexpr LogMsgCategory MCstats(MSGAUD_operator, MSGCLS_progress);
-constexpr LogMsgCategory MCoperatorInfo(MSGAUD_operator, MSGCLS_information);
+constexpr LogMsgCategory MCdisaster(MSGAUD_all, MSGCLS_disaster, FatalMsgThreshold);
+constexpr LogMsgCategory MCuserError(MSGAUD_user, MSGCLS_error, ErrMsgThreshold);
+constexpr LogMsgCategory MCoperatorError(MSGAUD_operator, MSGCLS_error, ErrMsgThreshold);
+constexpr LogMsgCategory MCinternalError(MSGAUD_programmer, MSGCLS_error, AllMsgThreshold);
```

Comment:
Internal logging this should have a high default threshold, could perhaps share default threshold of another one, e.g. DebugMsgThreshold

Reply:
Agreed, I think we're in synch but using contrary terms. Internal errs are assigned the lowest threshold. In affect, internal errors will always be streamed.

High threshold == only output if the admin has requested logging at a high level

Reply:
yes, internal should only be output if admin has requested logging at a high level.

Reply:
I am not sure that's true.
I thought internal, is akin to programmer, i.e. only of interest to devs, and so only if logging level turned up.
Most of the IERRLOG with this category, do not seem useful to most, so should not always be output afaics.

Reply:
Well, Internal Error means just that, and those should always be reported. 
I do see a lot instances of IERRLOG that seem to contain debug-like messages... I consider that to be a misuse of the IERRLOG function

Reply:
As per discussion offline, in JIRA and with @shamser.
The default override level fo the MC\*internal* categories, appears to have always been wrong.
i.e. setting it to the absolute highest priority and equal to MCdisaster can't be right.

I think we should get rid of AllMsgThreshold, and change MCinternalError to use same default threshold as other error categories, i.e.:
```
constexpr LogMsgCategory MCinternalError(MSGAUD_programmer, MSGCLS_error, ErrMsgThreshold);
```

@shamser - does that make sense?